src/monitors.ts

```diff
@@ -244,16 +246,24 @@ export function replaceCriticalThreshold(query: string, criticalThreshold: numbe
 }
 
 /**
- * Helper function that interpolates template variables, e.g. replaces template
- * variable `$functionName` with its default value `{{functionname.name}}`.
- * @param templateVariables - Template variables defined in the monitor JSON
- * @param name - The string that may contain uninterpolated template variables,
+ * Helper function that interpolates template variables (if any), e.g.
+ * replaces template variable `$functionName` with its default value
+ * `{{functionname.name}}`. Template variables with no default value will be
+ * ignored.
+ * @param recommendedMonitors  - recommended monitors
+ * @param baseString - The string that may contain uninterpolated template variables,
  *        e.g. "High Error Rate on $functionName in $regionName for $awsAccount"
  * @returns Interpolation result, e.g. "High Error Rate on {{functionname.name}} in {{region.name}} for {{aws_account.name}}"
  */
-function interpolateTemplateVariables(templateVariables: TemplateVariable[], name: string): string {
-  templateVariables.forEach((templateVariable) => {
-    name = name.replace("$" + templateVariable.name, templateVariable.defaults[0]);
-  });
-  return name;
+function interpolateTemplateVariables(recommendedMonitor: ServerlessMonitor, baseString: string): string {
+  const templateVariables = recommendedMonitor.templateVariables;
+  let interpolatedString = baseString;
+  if (templateVariables !== undefined && templateVariables.length > 0) {
+    templateVariables.forEach((templateVariable) => {
```

Comment:
***Nit:*** Just a small suggestion here. Since this is typescript, this code can be simplifed to just the below
```
    let interpolatedString = recommendedMonitor.name;
    recommendedMonitor.templateVariables?.forEach((templateVariable) => {
```